### PR TITLE
[dynamo] Add is_compiling for dynamo

### PR DESCRIPTION
**Pull request by mlazos**
`is_tracing` returns True during dynamo tracing and False when run in Eager

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/variables/functions.py [hunk 1]
```diff
@@ -193,6 +193,11 @@ def call_function(
                 tx, self.fn, self.get_name(), options, args, kwargs
             )
 
+        if self.fn == is_tracing:
```
Comment (jansel) [inline torch/_dynamo/variables/functions.py:196]: You should be able to get this behavior with:
https://github.com/pytorch/pytorch/blob/f28927e9c4694819f95c5dda2b111b0ff6dbd606/torch/_dynamo/config.py#L54

File: torch/_dynamo/variables/builder.py [hunk 2]
```diff
@@ -421,6 +422,10 @@ def index_source(key):
                 value,
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
+        elif value == is_tracing:
```
Comment (jansel) [inline torch/_dynamo/variables/builder.py:425]: You should be able to get this behavior with:
https://github.com/pytorch/pytorch/blob/f28927e9c4694819f95c5dda2b111b0ff6dbd606/torch/_dynamo/allowed_functions.py#L192

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_dynamo/variables/functions.py (hunk 1 above)
Comment (mlazos) [reply torch/_dynamo/variables/functions.py:196]: Thanks, this looks way simpler

File: torch/_dynamo/variables/builder.py (hunk 2 above)
Comment (mlazos) [reply torch/_dynamo/variables/builder.py:425]: Wait what about skipfiles though? The reason I did this was to override skipfiles

**Comment by mlazos** [comment]
> Is `is_tracing` the right word? We are not really tracing, but rather analyzing the bytecode.

I'm okay with other names, I thought is_tracing was clearest though. What do you suggest?

**Comment by soumith** [comment]
maybe `is_compiling`?

**Comment by mlazos** [comment]
> maybe `is_compiling`?

Sure, sgtm

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**